native/src/routes/jpal-tracking/JpalTracking.js

```diff
@@ -0,0 +1,141 @@
+import styled from 'styled-components/native'
+import LayoutContainer from '../../modules/layout/containers/LayoutContainer'
+import withTheme from '../../modules/theme/hocs/withTheme'
+import Caption from '../../modules/common/components/Caption'
+
+const ThemedText = styled.Text`
+  display: flex;
+  text-align: left;
+  color: ${props => props.theme.colors.textColor};
+  font-family: ${props => props.theme.fonts.decorativeFontRegular};
+  padding: 10px 0;
+`
+
+const ErrorText = styled.Text`
+  color: red;
+  font-weight: bold;
+  padding: 10px 0;
+`
+
+const DescriptionContainer = styled.View`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  padding: 15px 0 5px;
+`
+
+const Input = styled(TextInput)`
+  padding: 15px;
+  border-width: 1px;
+  border-color: ${props => props.theme.colors.themeColor};
+  text-align-vertical: top;
+  height: 50px;
+`
+
+export type PropsType = {|
+  theme: ThemeType,
+  route: RoutePropType<JpalTrackingRouteType>,
+  navigation: NavigationPropType<JpalTrackingRouteType>
+|}
+
+const errorDisplayTime = 5000
+
+const JpalTracking = (props: PropsType, state: StateType) => {
+  const appSettings = useMemo(() => new AppSettings(), [])
+  const [settings, setSettings] = useState(defaultSettings)
+  const [settingsLoaded, setSettingsLoaded] = useState(false)
+  const [settingsUpdated, setSettingsUpdated] = useState(false)
+  const [displayedTrackingCode, setDisplayedTrackingCode] = useState('')
+  const [displayedTrackingEnabled, setDisplayedTrackingEnabled] = useState(false)
+  const [error, setError] = useState(false)
+  const routeTrackingCode = props.route.params.trackingCode
+
+  const { t } = useTranslation(['settings', 'error'])
+
+  useEffect(() => {
+    if (routeTrackingCode) {
+      appSettings.setJpalTrackingCode(routeTrackingCode)
+    }
+  }, [appSettings, routeTrackingCode])
+
+  useEffect(() => {
+    const loadSettings = async () => {
+      try {
+        const loadedSettings = await appSettings.loadSettings()
+        setSettingsLoaded(true)
+        setSettingsUpdated(false)
+        setSettings(loadedSettings)
+        setDisplayedTrackingCode(loadedSettings.jpalTrackingCode)
+        setDisplayedTrackingEnabled(loadedSettings.jpalTrackingEnabled)
+        setTimeout(() => setError(false), errorDisplayTime)
+      } catch (e) {
+        // setError(true)
+      }
+    }
+    loadSettings()
+  }, [appSettings, settingsUpdated])
+
+  const toggleTrackingEnabled = () => {
+    setDisplayedTrackingEnabled(!displayedTrackingEnabled)
+    appSettings.setJpalTrackingEnabled(!settings.jpalTrackingEnabled).catch(() => {
+      setError(true)
+      setSettingsUpdated(true)
+    })
+  }
+
+  const setTrackingCode = (value: string) => {
+    setDisplayedTrackingCode(value)
+    appSettings.setJpalTrackingCode(value).catch(() => {
+      setError(true)
+      setSettingsUpdated(true)
+    })
+    props.navigation.setParams({ trackingCode: value })
```

Comment:
I think it is enough to just set the tracking code from the navigation params initially. No need to keep the tracking code in the navigation params consistent afterwards.